core/types/src/report.rs

```diff
@@ -33,6 +34,22 @@ impl_newtype_for_bytestruct! {
     ReportData, sgx_report_data_t, SGX_REPORT_DATA_SIZE, d;
 }
 
+/// There are times when only part of [`ReportData`] is of interest. [`BitAnd`]
+/// allows clients to mask off the parts of [`ReportData`] that are not of
+/// interest.
+impl BitAnd for ReportData {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        // NB: Due to use in verification, this must be constant time.
+        let mut output = Self::default();
+        for (i, (a, b)) in self.0.d.iter().zip(rhs.0.d.iter()).enumerate() {
```

Comment:
Wow, this is _dense_. I think I understand what's happening here but I think a simple `for i in range` type loop would be a lot more readable.

Reply:
Do you have a suggested implementation that you're thinking of?
Just for sanity I did this https://godbolt.org/z/W8ae5rPM8 to provide a bit more confidence the compiler won't care either way (if one were micro-optimizing), but also to show constant time ass.

I don't have a preference one way or the other. (though I have some scars from embedded compilers that would re-compute the array offset even in release :()
```rust
        for (i, o) in output.0.d.iter_mut().enumerate() {
            *o = self.0.d[i] & rhs.0.d[i];

        }
```

Reply:
Going to merge for now and we can discuss how dense we want to get in iterators moving forward